peer/peer.go

```diff
@@ -1406,7 +1422,7 @@ out:
 		// Read a message and stop the idle timer as soon as the read
 		// is done.  The timer is reset below for the next iteration if
 		// needed.
-		rmsg, buf, err := p.readMessage()
+		rmsg, buf, err := p.readMessage(p.wireEncoding)
```

Comment:
Yep, you're right!  I forgot that the refactor did the negotiation before starting the `inHandler`.  Previously the version message was received via the `inHandler`, so in that case it would have been.  

However, due the refactor, it is indeed set via `readRemoteVersionMsg` which is only called before the goroutine is launched via `negotiateInboundProtocol` and `negotiateOutboundProtocol`.